test/test_subscription.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import json
+from nostr.filter import Filter, Filters
+from nostr.message_type import ClientMessageType
+from nostr.subscription import Subscription
+
+
+def test_subscription_id():
+    """
+        check that subscription contents dump to JSON and load
+        back to Python with expected types
+    """
+    filters = Filters([Filter()])
+    id = 123
+    
+    with pytest.raises(TypeError) as e:
+        subscription = Subscription(id=id, filters=filters)
+    assert "Argument 'id' must be of type str" == str(e)
```

Comment:
nit: add a new line after this assert. My fault, I should've added this in my suggestion.